internal/storage/rocksdb/store.go

```diff
@@ -505,6 +566,22 @@ func toByteArray(value *gorocksdb.Slice) []byte {
 	return res
 }
 
+const tsLength = 8
+const endOfWorld = 253402300799 // 9999-12-31 23:59:59
+
+func getExpireTs(valueWithTtl []byte) (int64, error) {
+	l := len(valueWithTtl)
+	if l < tsLength {
+		return 0, fmt.Errorf("invalid_length")
+	}
+	ts := valueWithTtl[l-tsLength:]
+	i := binary.LittleEndian.Uint64(ts)
+	if i > 0 && i < endOfWorld {
+		return int64(i), nil
```

Comment:
What if this value represents a genuine timestamp and we interpret here as an expiry ? Should we have a byte mark to separate the value and TS portions ?

Reply:
Added a byte marker for this. This should be resolved not be an issue until the `endOfWorld` TS.